File: LLM-Translations/GPT-4/codenet/Java/Python/s283652523.py
import sys
from io import StringIO
import atexit

def main():
    N = int(input())
    A = list(map(int, input().split()))
    
    ans = 0
    MOD = 1000000007
    sum_ = A[N-1]
    for i in range(N - 2, -1, -1):
        ans += (sum_ * A[i]) % MOD
        ans %= MOD
        sum_ += A[i]
        sum_ %= MOD
    print(ans)

if __name__ == '__main__':
    main()

File: LLM-Translations/GPT-4/codenet/Java/C/s283652523.c
#include <stdio.h>
#include <stdlib.h>

#define MOD 1000000007

long long int sum_mod(long long int a, long long int b) {
    return (a + b) % MOD;
}

long long int product_mod(long long int a, long long int b) {
    return (a * b) % MOD;
}

void solve(int N, int *A, long long int *ans) {
    long long int sum = A[N - 1];
    *ans = 0;
    for (int i = N - 2; i >= 0; i--) {
        *ans = sum_mod(*ans, product_mod(sum, A[i]));
        sum = sum_mod(sum, A[i]);
    }
}

int main() {
    int N;
    scanf("%d", &N);

    int *A = (int *) malloc(N * sizeof(int));
    for (int i = 0; i < N; i++) {
        scanf("%d", &A[i]);
    }

    long long int ans;
    solve(N, A, &ans);

    printf("%lld\n", ans);

    free(A);

    return 0;
}
// 
